dataprofiler/profilers/numerical_column_stats.py

```diff
@@ -574,6 +590,41 @@ def _correct_bias_kurtosis(match_count, biased_kurtosis):
                    (biased_kurtosis + 3) - 3 * (match_count - 1))
         return kurtosis
 
+    def _estimate_mode_from_histogram(self):
+        """
+        Estimates the mode of the current data using the
+        histogram. If there are multiple modes, returns
+        K of them (where K is defined in options given, but
+        5 by default)
+
+        :return: The estimated mode of the histogram
+        :rtype: list(float)
+        """
+        if not self._has_histogram:
+            return np.nan
+
+        bin_counts = self._stored_histogram['histogram']['bin_counts']
+        bin_edges = self._stored_histogram['histogram']['bin_edges']
+
+        # Get the K bin(s) with the highest frequency (one-pass):
+        cur_max = -1
+        highest_idxs = []
+        count = 0
+        for i in range(0, len(bin_counts)):
+            if bin_counts[i] > cur_max:
+                highest_idxs = [i]
+                cur_max = bin_counts[i]
+                count = 1
```

Comment:
Not critical, but I would leave comments explaining why the `count` and `highest_idxs` are being reset here, since I thought `_top_k_modes` meant the top `k` most frequent data points, not the top `k` data points of the (singular) highest frequency.